docs/source/model/python_model.rst

```diff
@@ -0,0 +1,437 @@
+
+.. code-block:: python
+
+    import mlflow
+
+
+    class MyModel(mlflow.pyfunc.PythonModel):
+        def predict(self, model_input: list[str], params=None) -> list[str]:
+            return model_input
+
+
+
+Option 2: Define a callable
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
+An alternative way to log a PythonModel is to define a callable that **takes a single argument** and returns a prediction. This callable can be
+logged as a PythonModel by passing it to ``mlflow.pyfunc.log_model``.
+
+.. tip::
+    Starting from MLflow 2.20.0, you can use @pyfunc decorator on the callable to enable data validation on the input based on the type hints.
+    Check :ref:`type hint usage in PythonModel <type-hint-usage-in-pythonmodel>` for more details.
+
+.. code-block:: python
+
+    def predict(model_input: list[str]) -> list[str]:
+        return model_input
+
+Log the model
+^^^^^^^^^^^^^
+PythonModel should be logged as python_function flavor using :py:func:`mlflow.pyfunc.log_model`.
+
+.. code-block:: python
+
+    import mlflow
+
+    with mlflow.start_run():
+        model_info = mlflow.pyfunc.log_model(
+            artifact_path="model",
+            python_model=MyModel(),
+            input_example=input_example,
+        )
+
+
+Validate the model before deployment
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+After logging the model, you can validate the model by loading the model and running predictions. In addition, it's highly recommended to use 
+:py:func`mlflow.models.predict` API to validate the model dependencies. Check `MLflow Model Validation <../models.html#validate-models-before-deployment>`_
+for more details.
+
+.. code-block:: python
+    
+    import mlflow
+
+    model = mlflow.pyfunc.load_model(model_info.model_uri)
+    model.predict(["hello", "world"])
+
+    mlflow.models.predict(
+        model_uri=model_info.model_uri,
+        input_data=["a", "b", "c"],
+        env_manager="uv",
+    )
+
+Deploy the model
+^^^^^^^^^^^^^^^^
+Final step to use your model in production is to deploy it. Follow `MLflow Model Deployment <../models.html#deployment-to-custom-targets>`_ guide to deploy the model.
+
+.. _type-hint-usage-in-pythonmodel:
+
+Type hint usage in PythonModel
+------------------------------
+Starting from MLflow 2.20.0, type hints support for PythonModel is introduced. You can use type hints to define the input and output types of the model.
+It introduces the following benefits:
+
+    - **Data validation**: MLflow validates the input data based on the type hints defined in the model. No matter if the model is a PythonModel instance or a loaded PyFunc model, the input data is consistently validated.
+    - **Type hint inference**: MLflow infers the input and output schema of the model based on the type hints defined in the model, and sets as the model signature.
+
+Supported type hints
+^^^^^^^^^^^^^^^^^^^^
+**Type hint used in the PythonModel must be list[...]** because PythonModel's predict function expects batch input data.
```

Comment:
```suggestion
**Type hints used within the PythonModel's input signature must be of type list[...]** because PythonModel's predict function expects batch input data.
```